src/protocols/LoRaWAN/LoRaWAN.cpp

```diff
@@ -277,7 +280,41 @@ int16_t LoRaWANNode::restoreChannels() {
   }
   return(RADIOLIB_ERR_NONE);
 }
-#endif
+
+void LoRaWANNode::clearSession() {
+  Module* mod = this->phyLayer->getMod();
+  uint8_t zeroes[RADIOLIB_AES128_BLOCK_SIZE] = { 0 };
+  mod->hal->writePersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_APP_S_KEY_ID), zeroes, RADIOLIB_AES128_BLOCK_SIZE);
+  mod->hal->writePersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_FNWK_SINT_KEY_ID), zeroes, RADIOLIB_AES128_BLOCK_SIZE);
+  mod->hal->writePersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_SNWK_SINT_KEY_ID), zeroes, RADIOLIB_AES128_BLOCK_SIZE);
+  mod->hal->writePersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_NWK_SENC_KEY_ID), zeroes, RADIOLIB_AES128_BLOCK_SIZE);
+}
+
+bool LoRaWANNode::isActiveSession() {
+  Module* mod = this->phyLayer->getMod();
+  this->devAddr = mod->hal->getPersistentParameter<uint32_t>(RADIOLIB_EEPROM_LORAWAN_DEV_ADDR_ID);
+  mod->hal->readPersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_APP_S_KEY_ID), this->appSKey, RADIOLIB_AES128_BLOCK_SIZE);
+  mod->hal->readPersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_FNWK_SINT_KEY_ID), this->fNwkSIntKey, RADIOLIB_AES128_BLOCK_SIZE);
+  mod->hal->readPersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_SNWK_SINT_KEY_ID), this->sNwkSIntKey, RADIOLIB_AES128_BLOCK_SIZE);
+  mod->hal->readPersistentStorage(mod->hal->getPersistentAddr(RADIOLIB_EEPROM_LORAWAN_NWK_SENC_KEY_ID), this->nwkSEncKey, RADIOLIB_AES128_BLOCK_SIZE);
```

Comment:
I don't think we should be overwriting the current keys just to test if a saved session exists. I'm not sure it actually breaks something, but it seems like a major side effect. If this method is really ment to just test the session exists, then it should save the keys into a local buffer.

Also, does this still work if the persistent storage is not initialized to 0x00s, but to 0xFFs?

Reply:
> I don't think we should be overwriting the current keys just to test if a saved session exists. I'm not sure it actually breaks something, but it seems like a major side effect. If this method is really ment to just test the session exists, then it should save the keys into a local buffer.

In itself a fair shout - I cannot immediately come up with a possible breaking situation, but safer with separate buffers. Will fix.

> Also, does this still work if the persistent storage is not initialized to 0x00s, but to 0xFFs?

This function would not be called unless the mode and checksum are also verified, which would both quickly fail if all is set to 0xFF. So I am not worried about that. If we still want it to be safe, I can put some extra thought into it..

Reply:
> This function would not be called unless the mode and checksum are also verified, which would both quickly fail if all is set to 0xFF. So I am not worried about that. If we still want it to be safe, I can put some extra thought into it..

I suppose it is private, so we have full control over when it gets called. No need to dig into the unitialized case any further, thanks for the clarification.